### ISSUE TITLE
Allow suspending a sale before adding a payment

### ISSUE THREAD

**Issue by daN4cat** [jekkos/opensourcepos#180]
Currently you can suspend a sale only after having added a payment, which is a bit of a non-sense as most of the times you would suspend a sale before a payment in case a customer wants to buy more.
That will happen before any payment.

**Comment by odiea** [comment]
My thoughts exactly.

**Comment by daN4cat** [comment]
Yes, I copy your point as separate issue. I prefer this way instead of "one-catch-all" :-)

**Comment by odiea** [comment]
Ok. I did not want to muck up the issues with multiple things I have noticed or wanted to change.

**Comment by daN4cat** [comment]
This is not so trivial as it might look like. It needs some more understanding of the flow before touching it.

**Comment by odiea** [comment]
Thanks and Good luck on that if you decide to tackle that job.

**Comment by daN4cat** [comment]
OK, but then you show the complete button straight away without checking if any payment is added.

**Comment by daN4cat** [comment]
Just checked, ok the complete button is under the if ($payments_cover_total) condition, but if you suspend the sale and then unsuspend you get the button back even if there is no payment.

**Comment by daN4cat** [comment]
I had a look into this and I think I managed to make the change properly.

I prefer to do not open the part under the if(count($payments) > 0) and I moved the suspend button on the right side of Cancel Sale as that looks to me more suitable and consistent with the flow.

If you don't add a payment you either want to cancel or suspend. That's the logic.

**Comment by daN4cat** [comment]
OK, I pushed the change to my master and I just tested few cases. More needs to be checked.

**Comment by odiea** [comment]
Yes indeed that button works much better there.

**Comment by daN4cat** [comment]
:+1:
